### DISCUSSION TITLE
Declare and use local variables per connection

### DISCUSSION THREAD

**Discussion by lzaiko** [uNetworking/uWebSockets.js#380] [Q&A] [answered]
Hi Alex,

I’m having an issue to declare a local variable **foo**  (see below) which is only visible inside the socket connection handlers (_open_, _message_, _close_).

```
cost port = 9000;
const app = uWS.App()
          .ws('/', {
              maxPayloadLength: 1024 * 1024,
              idleTimeout: 15,
              open: (ws, req) => {
                  foo = true;
              },
              message: (ws, message, isBinary) => {
                 if(foo) {
                   // do something with the message
                  }
              }
          }).listen(port, token => {
             console.log(`WebSocket server listening on: ${port}`);
          });
});
```
The variable **foo** can’t be declared globally, otherwise it will be visible for all the connections (total mess).

I checked the examples but couldn’t find a solution to my problem.
Help appreciated.

**Comment by hst-m** [comment, marked as answer]
You've got an extra set of `})` at the end that needs to be removed, and `open` event does not have `req` so you need to remove that, and if you want to share something across `open` and `message` events you should put it in the `ws` object
```javascript
open:ws=>{
   ws.foo=true
},
message:(ws,message,isBinary)=>{
   console.log(ws.foo)
}
```

**Comment by lzaiko** [comment]
Thanks a mill.
